parsec/core/mountpoint/winfsp_operations.py

```diff
@@ -190,27 +193,35 @@ def stat_to_winfsp_attributes(stat):
     return attributes
 
 
-def handle_error(func):
+R = TypeVar("R")
+P = ParamSpec("P")
+
+
+def handle_error(func: Callable[P, R]) -> Callable[..., R]:
```

Comment:
```suggestion
def handle_error(func: Callable[P, R]) -> Callable[Concatenate[BaseFileSystemOperations, OpenedFile | OpenedFolder | str, P], R]:
```

Using concatenate would not work here ?